src/main/java/com/yelp/nrtsearch/server/luceneserver/QueryNodeMapper.java

```diff
@@ -660,8 +666,131 @@ private SpanQuery getSpanQuery(
             new Term(protoSpanTermQuery.getField(), protoSpanTermQuery.getTextValue()));
+      case SPANMULTITERMQUERY:
+        com.yelp.nrtsearch.server.grpc.SpanMultiTermQuery protoSpanMultiTermQuery =
+            protoSpanQuery.getSpanMultiTermQuery();
+        return getSpanMultiTermQueryWrapper(protoSpanMultiTermQuery, state);
       default:
         throw new IllegalArgumentException("Unsupported Span Query: " + protoSpanQuery);
     }
   }
+
+  private SpanMultiTermQueryWrapper getSpanMultiTermQueryWrapper(
+      com.yelp.nrtsearch.server.grpc.SpanMultiTermQuery protoSpanMultiTermQuery, IndexState state) {
+
+    com.yelp.nrtsearch.server.grpc.SpanMultiTermQuery.WrappedQueryCase wrappedQueryCase =
+        protoSpanMultiTermQuery.getWrappedQueryCase();
+
+    switch (wrappedQueryCase) {
+      case WILDCARDQUERY:
+        WildcardQuery wildcardQuery =
+            new WildcardQuery(
+                new Term(
+                    protoSpanMultiTermQuery.getWildcardQuery().getField(),
+                    protoSpanMultiTermQuery.getWildcardQuery().getText()));
+        wildcardQuery.setRewriteMethod(
+            getRewriteMethod(
+                protoSpanMultiTermQuery.getWildcardQuery().getRewrite(),
+                protoSpanMultiTermQuery.getWildcardQuery().getRewriteTopTermsSize()));
+        return new SpanMultiTermQueryWrapper<>(wildcardQuery);
+      case FUZZYQUERY:
+        FuzzyQuery fuzzyQuery = getFuzzyQuery(protoSpanMultiTermQuery);
+        return new SpanMultiTermQueryWrapper<>(fuzzyQuery);
+      case PREFIXQUERY:
+        Query prefixQuery = getPrefixQuery(protoSpanMultiTermQuery.getPrefixQuery(), state);
+        return new SpanMultiTermQueryWrapper<>((MultiTermQuery) prefixQuery);
+      case REGEXPQUERY:
+        RegexpQuery regexpQuery = getRegexpQuery(protoSpanMultiTermQuery);
+        return new SpanMultiTermQueryWrapper<>(regexpQuery);
+      case TERMRANGEQUERY:
+        TermRangeQuery termRangeQuery =
+            getTermRangeQuery(protoSpanMultiTermQuery.getTermRangeQuery());
+        return new SpanMultiTermQueryWrapper<>(termRangeQuery);
+      default:
+        throw new IllegalArgumentException(
+            "Unsupported Span Multi Query Term Wrapper: " + protoSpanMultiTermQuery);
+    }
+  }
+
+  private static FuzzyQuery getFuzzyQuery(
+      com.yelp.nrtsearch.server.grpc.SpanMultiTermQuery protoSpanMultiTermQuery) {
+    com.yelp.nrtsearch.server.grpc.FuzzyQuery protoFuzzyQuery =
+        protoSpanMultiTermQuery.getFuzzyQuery();
+    Term term = new Term(protoFuzzyQuery.getField(), protoFuzzyQuery.getText());
+
+    int maxEdits =
+        protoFuzzyQuery.getMaxEdits() == FuzzyQuery.defaultMaxEdits
+            ? FuzzyQuery.defaultMaxEdits
+            : protoFuzzyQuery.getMaxEdits();
+
+    int prefixLength =
+        protoFuzzyQuery.getPrefixLength() == FuzzyQuery.defaultPrefixLength
+            ? FuzzyQuery.defaultPrefixLength
+            : protoFuzzyQuery.getPrefixLength();
```

Comment:
I'm not sure I understand the defaults logic here. If the set value is the default value, use the default value, else use the set value. Wouldn't this just be always using the set value? I don't think this matches the doc strings behavior.

Reply:
The default value for this is `0`. So `protoFuzzyQuery.getPrefixLength()` always returns the correct value, even if no value is set for it. If the default value is anything other than `0` using the set value will not use the correct default value. I'm not too sure if `defaultPrefixLength` would ever change in future versions of Lucene. But I think it's a good idea to keep the logic this way so that the default value won't be ever overriden.

Reply:
Sure, I changed all fields with default values to `optional`.